pkg/resources/counts/buffer_test.go

```diff
@@ -0,0 +1,95 @@
+package counts
+
+import (
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_countsBuffer(t *testing.T) {
+	tests := []struct {
+		name           string
+		numInputEvents int
+		overrideInput  map[int]int // events whose count we should override. Don't include an event >= numInputEvents
+	}{
+		{
+			name:           "test basic input",
+			numInputEvents: 1,
+		},
+		{
+			name:           "test basic multiple input",
+			numInputEvents: 3,
+		},
+		{
+			name:           "test basic input which is overriden by later events",
+			numInputEvents: 3,
+			overrideInput: map[int]int{
+				1: 17,
+			},
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			debounceDuration = 100 * time.Millisecond
+			countsChannel := make(chan Count, 100)
+			outputChannel := countsBuffer(countsChannel)
+
+			// first event is not buffered, so get it out of the way to test the debounce
+			countsChannel <- Count{
+				ID:     "count",
+				Counts: map[string]ItemCount{"test": createItemCount(1)},
+			}
+			<-outputChannel
```

Comment:
Nit: 
You could use `receiveWithTimeout` here to verify that the first result is returned immediately. This would be primarily to stop someone from introducing a regression if buffer is ever updated.

Reply:
I'll have to re-implement it since this in a different package than counts_test.go (I can test public interfaces with the other one but not with this one), but I think that's ok since it's just a helper/utility method.

Reply:
Re-implemented this. I've also changed the other part of the test to use this method too.